recipes/mathter/all/conanfile.py

```diff
@@ -0,0 +1,49 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class MathterConan(ConanFile):
+    name = "mathter"
+    license = "MIT"
+    homepage = "https://github.com/petiaccja/Mathter"
+    url = "https://github.com/conan-io/conan-center-index/"
+    description = "Powerful 3D math and small-matrix linear algebra library for games and science."
+    topics = ("game-dev", "linear-algebra", "vector-math", "matrix-library")
+    no_copy_source = True
+    settings = "compiler"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": 10,
+            "clang": 6,
+            "gcc": 7,
+            "Visual Studio": 16,
+        }
+
+    def configure(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "17")
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version:
+            if tools.Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration("mathter requires C++17, which your compiler does not support.")
+        else:
+            self.output.warn("mathter requires C++17. Your compiler is unknown. Assuming it supports C++17.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("Mathter-" + self.version, self._source_subfolder)
+            
+    def package(self):
+        self.copy("*.hpp", dst=os.path.join("include", "Mathter"), src=os.path.join(self._source_subfolder, "Mathter"))
+        self.copy("*.natvis", dst=os.path.join("include", "Mathter"), src=os.path.join(self._source_subfolder, "Mathter"))
```

Comment:
```suggestion
        self.copy("*.natvis", dst=os.path.join("include", "Mathter"), src=os.path.join(self._source_subfolder, "Mathter"))
```

It is for debugging or what? If yes, can we remove? We don't package pdb or any other debug stuff, because 99% people download only the package and 1% they want to debug something.

Reply:
Good catch!

```suggestion
```

https://docs.microsoft.com/en-us/visualstudio/debugger/create-custom-views-of-native-objects?view=vs-2019#BKMK_Using_Natvis_files

Reply:
It's to help debugging applications that use Mathter. When writing a game, you often want to see the value of the matrices in the debugger to understand what your code is doing. By default, a vector is displayed by it's internal implementation, which is a ```union {{x=1,y=2,z=3}, data, simd, xx, xy, yx, yy, ...};```, so it's quite difficult to read. The pretty printer displays the same vector like ```[1, 2, 3]```, which makes developing with Mathter much more pleasant.

The point is, it's a very useful piece of the library, and I'd like to include it in the package. That said, I'm not sure what would be the best way to distribute it with conan. This file works with Visual Studio, it can be added to the solution or included as a source in CMake, that's why it makes sense to me to put it next to the source files. If there is any better way to handle this with conan, I'm open to suggestions.

Reply:
Okay, as it's a xml file only and not a binary like .pdb, it looks okay. But now I'm curious how do you debug without Visual Studio?

Reply:
Well, you can still look at stuff, there is just much more clutter. Anyways, I want to add pretty printers for GDB and other platforms, but it was never a high priority.